operators/pkg/controllers/workspace_controller.go

```diff
@@ -42,16 +46,34 @@ type WorkspaceReconciler struct {
 // Reconcile reconciles the state of a workspace resource
 func (r *WorkspaceReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	ctx := context.Background()
-
 	var ws crownlabsv1alpha1.Workspace
+	accToken := r.KcToken.AccessToken
+	targetRealm := r.KcTargetRealm
+	targetClientID, err := getClientID(ctx, r.KcClient, accToken, targetRealm, r.KcTargetClient)
```

Comment:
I would move this to the "main", to avoid performing this request at every reconcile